src/app/core/campaign/campaign.service.ts

```diff
@@ -1,16 +1,24 @@
 import { Injectable } from '@angular/core';
+import { Store, select } from '@ngrx/store';
 import { Observable, of } from 'rxjs';
-import { map, switchMap, catchError } from 'rxjs/operators';
+import { map, switchMap, catchError, find } from 'rxjs/operators';
 import { HalDoc } from 'ngx-prx-styleguide';
 import { AuguryService } from '../augury.service';
 import { CampaignState, Campaign, Flight, FlightState } from './campaign.models';
+import { selectCampaignDoc } from '../../campaign/store/selectors';
 
 @Injectable()
 export class CampaignService {
-  private campaignDoc: HalDoc;
-  private flightDocs: { [id: number]: HalDoc };
+  campaignDoc$: Observable<HalDoc>;
```

Comment:
This service uses the HalDoc to later make updates, delete/destroy, or create flights on a campaign. Instead of keeping the HalDoc here in the service and relying on it having been set in order to create flights or update the campaign, it's been moved to the store and gets saved each campaign get/put/post. This way, when creating flights on a new campaign, we can wait on that HalDoc Observable to have been set in order to create the flights.

Reply:
+1 good call.